### PR TITLE
Server improvements from pr 4

### PR DESCRIPTION
**Pull request by jonas27**

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(exitFail)
+	}
+}
+
+func run(args []string) error {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	flags := flag.NewFlagSet(args[0], flag.ExitOnError)
+	var (
+		port = flags.String("port", ":8080", "The server port with colon")
```
Comment (avorima) [inline server/main.go:36]: address

File: server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	t.Parallel()
+	tests := []struct {
+		name     string
+		code     int
+		path     string
+		key      string
+		respBody string
+	}{
+		{name: "key exists", code: http.StatusOK, path: "/db", key: "test"},
+		{name: "key does not exist", code: http.StatusNotFound, path: "/", key: "test1"},
```
Comment (avorima) [inline server/server_test.go:23]: The test succeeds if you use the correct key. Seems like the routes aren't set up correctly

File: server/server_test.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+func TestPost(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		path     string
+		key      string
+		reqBody  string
+		code     int
+		response string
```
Comment (avorima) [inline server/server_test.go:88]: unused

File: server/server_test.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+func TestPut(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		path     string
+		key      string
+		reqBody  string
+		code     int
+		response string
```
Comment (avorima) [inline server/server_test.go:122]: unused

**Review by avorima — CHANGES_REQUESTED** [review]

File: server/main.go (hunk 1 above)
Comment (jonas27) [reply server/main.go:36]: yeah, I fixed it in the old PR and now I am stuck merging it up with every PR (included in current newest PR)

**Pull request closed without merging**